src/software/ai/move_rule/move_rule_manager_test.cpp

```diff
@@ -0,0 +1,162 @@
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::GOAL_US);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::GOAL_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::BALL_PLACEMENT_THEM);
+    correct_move_rules = std::set<MoveRule>(
+        {MoveRule::INFLATED_ENEMY_DEFENSE_AREA, MoveRule::ENEMY_ROBOTS_COLLISION,
+         MoveRule::HALF_METER_AROUND_BALL, MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+}
+
+TEST(MoveRuleTest, test_avoid_areas_from_game_start_or_us_states)
+{
+    GameState game_state;
+    MoveTactic move_tactic;
+    MoveRuleManager manager;
+    std::set<MoveRule> correct_move_rules;
+
+    game_state.updateRefboxGameState(RefboxGameState::FORCE_START);
+    correct_move_rules = std::set<MoveRule>({MoveRule::INFLATED_ENEMY_DEFENSE_AREA,
+                                             MoveRule::ENEMY_ROBOTS_COLLISION,
+                                             MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::DIRECT_FREE_US);
+    correct_move_rules = std::set<MoveRule>({MoveRule::INFLATED_ENEMY_DEFENSE_AREA,
+                                             MoveRule::ENEMY_ROBOTS_COLLISION,
+                                             MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::INDIRECT_FREE_US);
+    correct_move_rules = std::set<MoveRule>({MoveRule::INFLATED_ENEMY_DEFENSE_AREA,
+                                             MoveRule::ENEMY_ROBOTS_COLLISION,
+                                             MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
+
+    game_state.updateRefboxGameState(RefboxGameState::BALL_PLACEMENT_US);
+    correct_move_rules = std::set<MoveRule>({MoveRule::INFLATED_ENEMY_DEFENSE_AREA,
+                                             MoveRule::ENEMY_ROBOTS_COLLISION,
+                                             MoveRule::FRIENDLY_DEFENSE_AREA});
+    EXPECT_EQ(correct_move_rules, manager.getMoveRules(game_state, move_tactic));
```

Comment:
Could we split these tests into `test_move_rules_for_move_tactic_in_FORCE_START_game_state`, `test_move_rules_for_move_tactic_in_INDIRECT_FREE_US_game_state`, etc. since there's no interdependence between them?

If we're just trying to test getting the move rules for a given game state, I'd probably suggest just exposing the `getMoveRulesFromGameState` as a `public` and `static` member, then testing that directly. If we are in fact trying to test combinations of Tactic + GameState (which is _great_ we totally should), I'd probably suggest a parameterized test.

Reply:
Hmmmm, thinking about the above a bit more, I think testing Tactic + GameState combo's would be _way_ more valuable.

Reply:
Yeah we shouldn't expose `getMoveRulesFromGameState` just for testing